### DISCUSSION TITLE
set custom schema in createClientComponentClient from auth-helpers-nextjs

### DISCUSSION THREAD

**Discussion by BorisBarzotto** [supabase/supabase#16253] [Questions]
Hi!,
How i can set a custom schema there?
g_empresa works fine in public schema, but i need  g_empresa in another schema. so  how i can set that createClientComponentClient to know the name of the custom schema, the way it is it only looks for it if the table exists in public with no problems

![image](https://github.com/supabase/supabase/assets/112579911/9b70f6f1-0502-444c-887b-a0c79257dd4c)

**Comment by GaryAustin1** [comment]
Looks like with 1.8 of Postgrest-js when it gets rolled up to supabase-js you can just add .schema('my_special_schema') to individual calls rather than have to set it on a specific client.  Not a direct answer to how to set the client to use a specific schema for all calls in auth-helpers, but seems like for most cases will be a better approach.  https://github.com/supabase/postgrest-js/releases/tag/v1.8.0

**Reply by GaryAustin1** [reply]
Now released on latest Supabase-js.

**Comment by BorisBarzotto** [comment]
I wanted to continue using the auth-helper to work directly with cookies, but I find that it uses only the default public schema and there is no way to change it and I received a database full of custom schemas, with the new implementation of supabase-js and using createClient I have no problems, but I find it hard to set cookies manually, it is still something I did without problems with this dependency. Specifically I use Nextjs

**Reply by GaryAustin1** [reply]
So the new .schema() method won't work for you at all?  Or it won't work because auth-helpers is not using that supabase-js version yet?

**Reply by GaryAustin1** [reply]
Also you probably could use getSession to get the current access token and then call a generic supabase-js client with the Authorization header set to 'Bearer access_token' and that client would work as the user for database calls.

**Reply by silentworks** [reply]
PRs are welcomed for this feature in the [auth-helpers](https://github.com/supabase/auth-helpers).
